src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/FlowAnalysis/GlobalFlowStateDictionaryFlowOperationVisitor.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.GlobalFlowStateAnalysis;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines.AvoidMultipleEnumerations.FlowAnalysis
+{
+    using GlobalFlowStateDictionaryAnalysisData = DictionaryAnalysisData<AnalysisEntity, GlobalFlowStateDictionaryAnalysisValue>;
+    using GlobalFlowStateDictionaryAnalysisResult = DataFlowAnalysisResult<GlobalFlowStateDictionaryBlockAnalysisResult, GlobalFlowStateDictionaryAnalysisValue>;
+
+    internal abstract class GlobalFlowStateDictionaryFlowOperationVisitor : GlobalFlowStateDataFlowOperationVisitor<
+        GlobalFlowStateDictionaryAnalysisContext,
+        GlobalFlowStateDictionaryAnalysisResult,
+        GlobalFlowStateDictionaryAnalysisValue>
+    {
+        protected GlobalFlowStateDictionaryFlowOperationVisitor(
+            GlobalFlowStateDictionaryAnalysisContext analysisContext) : base(analysisContext, true)
+        {
+        }
+
+        public override GlobalFlowStateDictionaryAnalysisData GetEmptyAnalysisData()
+            => new();
+
+        public sealed override (GlobalFlowStateDictionaryAnalysisData output, bool isFeasibleBranch) FlowBranch(BasicBlock fromBlock, BranchWithInfo branch, GlobalFlowStateDictionaryAnalysisData input)
```

Comment:
This override can be removed if you address the previous comment in the base type.